script.js

```diff
@@ -0,0 +1,200 @@
+let webBrands = [];
+let brands=[];
+let resultbrands=[];
+let productNode;
+const proxyurl = "https://cors-anywhere.herokuapp.com/";
+const url = `https://wellspent-ethical.herokuapp.com/api/brands`;
+
+
+var linksList = document.querySelectorAll("[id^=product-]");
+linksList.forEach(function(product) {
+  const node = product.childNodes[0];
+  const label = node.getAttribute("aria-label");
+  let brand = label.toLowerCase();
+  webBrands.push(brand);
+});
+
+
+ // site that doesn’t send Access-Control-*
+fetch(proxyurl + url) // https://cors-anywhere.herokuapp.com/https://example.com
+  .then(resultbrandsonse => resultbrandsonse.text())
+  .then(contents => {
+    resultbrands = JSON.parse(contents);
+    brands = resultbrands;
+
+
+    const finalResult = brands.result.map((data)=>{
+    const result =   webBrands.filter((webBrand)=>{
+      const splitedBrand= data.BrandName.toLowerCase().split(" ")[0];
+      if(webBrand.indexOf(splitedBrand)==0){
+        return webBrand.includes(data.BrandName.toLowerCase())
+      }
+      })
+      return result.length > 0 ? [data.BrandName] : [];
+    });
+    const filteredBrand = Promise.all(finalResult)
+    .then((res)=>{
+        res.map((brand,index, array) => {
+          if(brand.length > 0){
+          linksList.forEach(function(product) {
+            const node = product.childNodes[0];
+            const label = node.getAttribute("aria-label");
+            let productNodeText = label.toLowerCase();
+            if(productNodeText.includes(brand[0].toLowerCase())){
+                product.style["border-color"] =resultbrands.colourMap[resultbrands.result[index].OverallScore];
+                product.classList.add('product');
+                var button = document.createElement("button");
+                button.innerText = "More information";
+                button.setAttribute("id", "btn");
+                product.appendChild(button);
+
+                var span =document.createElement("span");
+                var myModal =document.createElement("div");
+
+                if(button){
+                button.onclick = function() {
+                  myModal.classList.add('myModal');
+                  myModal.setAttribute("id", "myModal");
+                  var content =document.createElement("div");
```

Comment:
You can use `const` to create HTML elements using JS, you didn't reassign any of them multiple times.